src/entities/card/model/index.ts

```diff
@@ -8,13 +9,30 @@ export const $cardsCache = createStore<{ cache: Record<string, Card> }>({
   cache: {},
 });
 
+export const cardsSaveFx = attach({ effect: internalApi.cardsSave });
+export const cardsUnsaveFx = attach({ effect: internalApi.cardsUnsave });
+
 export const $cards = createStore<Card[]>([]);
+
+// @TODO It's bad practice to use global store. Will be fixed after BOX-250
+export const $favoritesIds = createStore<string[]>([]);
+
+export const changeFavorites = createEvent<string[]>();
+
+export const $favoritesCards = combine(
+  $favoritesIds,
+  $cardsCache,
+  (ids, { cache }) => ids.map((id) => cache[id]),
```

Comment:
У тебя может возникнуть ситуация, когда карточка в списке избранного ещё не попала в кеш. Тогда у тебя будет в сторе массив с `undefined`. И когда ты промаппишь массив в одно значение, то получишь пропуск обновления стора, либо падение в рантайме.

Решение 1: явно маппить в null. Тогда тайпскрипт подхватит тайпинги и ты сможешь промаппить значение в эффекторе и отловить неконсистентность

```suggestion
  (ids, { cache }) => ids.map((id) => cache[id] ?? null),
```

Решение 2: отфильтровать не найденное. Но тогда у тебя будет пустой массив, пока в кеше не появятся карточки, что не плохо, но иногда это не ожидаемое поведение

```suggestion
  (ids, { cache }) => ids.map((id) => cache[id]).filter(Boolean),
```

Reply:
fixed